tests/test_flowdiagnosticstool.cpp

```diff
@@ -344,7 +344,7 @@ BOOST_AUTO_TEST_CASE (OneDimCase)
         const auto tof = fwd.fd.timeOfFlight();
 
         BOOST_REQUIRE_EQUAL(tof.size(), cas.connectivity().numCells());
-        std::vector<double> expected = { 0.5, 1.5, 2.5, 3.5, 4.5 };
+        std::vector<double> expected = { 0.5, 1.5, 2.5, 3.5, 0.0 };
```

Comment:
I think this is okay, if slightly unintuitive in isolation.  In a real case one would probably not specify both sources and sinks in a single call to `compute{Injection,Production}Diagnostics()`, but here it probably does makes sense.